src/Dict.h

```diff
@@ -1504,6 +1533,9 @@ template <typename T> class Dictionary
 	uint32_t num_entries = 0;
 	uint32_t max_entries = 0;
 	uint64_t cum_entries = 0;
+	int64_t space_distance_samples = 0;
+	// how far the space is
```

Comment:
Is this is the space between two elements in the table? I'm not entirely sure what you mean by "space" in all of the changes.

Reply:
When a new item is to be inserted, it is inserted as the last element in its cluster (elements of the same bucket). However, if that spot is not empty (not a space), it has to relocate the existing element that occupies the spot before inserting itself.  Now the new element needs to find its new location. And the new location is the end of its own cluster. The chain effect continues until a space (empty spot) is found so that no new element needs to be relocated.

The space distance is the distance between this empty spot and the spot of the externally inserted element. 

This chain effect, (like recursion) causes a linear search until the next space is found in the table from bucket of the element.  When space is hard to find, it takes long time.

The reason the space if faraway is the load factor is just too large for the current hash distribution.

Reply:
Ah, so it's basically a factor of how long it takes for the table to rebalance after inserting an element that would initially be considered a collision?

Reply:
That's correct.